### PR TITLE
Add metadata for VLM blocks

### PR DESCRIPTION
**Pull request by PawelPeczek-Roboflow**
# Description

This PR adds:
* description for each task type of VLM and meaningful task names
* custom metadata that help to pick VLM blocks
* default tasks

List any dependencies that are required for this change.

## Type of change

Please delete options that are not relevant.

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [ ] This change requires a documentation update

## How has this change been tested, please provide a testcase or example of how you tested the change?

* tests still 🟢 

## Any specific deployment considerations

For example, documentation changes, usability, usage/costs, secrets, etc.

## Docs

-   [ ] Docs updated? What were the changes:

### CODE REVIEW

File: inference/core/workflows/core_steps/common/vlms.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    },
+    "region-proposal": {
+        "name": "Regions of Interest proposal",
+        "description": "Model proposes Regions of Interest (Bounding Boxes) in the image",
+    },
+    "structured-answering": {
+        "name": "Structured Output Generation",
+        "description": "Model produces JSON structure that you specify",
+    },
+}
```
Comment (grzegorz-roboflow) [inline inference/core/workflows/core_steps/common/vlms.py:83]: If in the future new LM will be added will it be compulsory to use this dict?

I'd vote for reducing code sharing between blocks

File: inference/core/workflows/core_steps/common/vlms.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
+VLM_TASKS_METADATA = {
+    "unconstrained": {
+        "name": "Open Prompt",
+        "description": "Use any prompt to generate a raw response",
```
Comment (EmilyGavrilenko) [inline inference/core/workflows/core_steps/common/vlms.py:4]: Open to description suggestions, best I could come up with

File: inference/core/workflows/core_steps/common/vlms.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    "detection-grounded-caption": {
+        "name": "Captioning of Bounding Box",
+        "description": "Model captions the object in the provided bounding box",
+    },
+    "detection-grounded-ocr": {
+        "name": "Text Recognition (OCR) for Bounding Box",
+        "description": "Model performs OCR on the text inside the provided bounding box",
+    },
+    "region-proposal": {
+        "name": "Regions of Interest proposal",
```
Comment (EmilyGavrilenko) [inline inference/core/workflows/core_steps/common/vlms.py:75]: What does region-proposal do?

File: inference/core/workflows/core_steps/common/vlms.py [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+        "description": "Model proposes Regions of Interest (Bounding Boxes) in the image",
+    },
+    "structured-answering": {
+        "name": "Structured Output Generation",
+        "description": "Model returns a JSON response with the specified fields",
```
Comment (EmilyGavrilenko) [inline inference/core/workflows/core_steps/common/vlms.py:80]: Can we add a `recommended_parser` field for each VLM_TASK that requires a parser block. That way we can dynamically show a tooltip to users 

Ex: `recommended_parser: JSON Parser`

Comment (PawelPeczek-Roboflow) [reply inference/core/workflows/core_steps/common/vlms.py:80]: 👍

File: inference/core/workflows/core_steps/common/vlms.py (hunk 3 above)
Comment (PawelPeczek-Roboflow) [reply inference/core/workflows/core_steps/common/vlms.py:75]: It looks like that: https://github.com/retkowsky/florence-2/blob/main/Florence-2%20-%20Advancing%20a%20Unified%20Representation%20for%20a%20Variety%20of%20Vision%20Tasks.ipynb

Model proposes all potential regions of interest in the image

File: inference/core/workflows/core_steps/common/vlms.py (hunk 4 above)
Comment (PawelPeczek-Roboflow) [reply inference/core/workflows/core_steps/common/vlms.py:80]: not sure shall we bound it to version of parser, probably not - announcing family only

Comment (PawelPeczek-Roboflow) [reply inference/core/workflows/core_steps/common/vlms.py:80]: the only problem I see is that not all VLMs from the very start needs to be possible to be parsed to specific type of object - for instance:
Florence has `detection-grounded-classification` which is not possible to be parsed to CLS object now, yet could be

Comment (PawelPeczek-Roboflow) [reply inference/core/workflows/core_steps/common/vlms.py:80]: ok, saw a suggestion in the Slack - yeah, I feel like it's better to decide per block - that would solve the problem I stated above

**Review by EmilyGavrilenko — APPROVED** [review]

**Review by grzegorz-roboflow — APPROVED** [review]

**Pull request merged**